File: stream/graph/A_newYearTransport.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    ios::sync_with_stdio(0); cin.tie(0);
    int n, t;
    cin >> n >> t;
    vector<int> a(n+1);
    for (int i = 1; i < n; i++) cin >> a[i]; 
    int cur = 1;
    do {
        cur = cur + a[cur];
        if (cur == t) {
            cout << "YES\n";
            return 0;
        }
    } while (cur < n);
    cout << "NO\n";
    return 0;
}

File: stream/graph/A_newYearTransport.py
n, t = map(int, input().split())
a = [0] + list(map(int,input().split()))
cur = 1
valid = False
while cur < n:
    cur = cur + a[cur]
    if cur == t:
        valid = True
        break
if valid:
    print("YES")
else:
    print("NO")
        